### DISCUSSION TITLE
Read two ADC (making a stereo source) and sending to A2DP

### DISCUSSION THREAD

**Discussion by fabioguerra-br** [pschatzmann/arduino-audio-tools#1046] [Q&A] [answered]
Hello Phil.

I'm using your base-adc-a2dp.ino as example, is possible make it stereo?
I need to read a analog stereo source (directly on ESP32 pins without external ADC) and send to bluetooth stereo phone.
Or have another idea to do this?

Thanks!

**Comment by pschatzmann** [comment, marked as answer]
To my knowledge, phones can not act as A2DP sink or did you find a solution for this ?
Can you connect from one Phone to an other ?

The AnalogAudioStream provides by default 2 channels. (The mono input is just provided on both channels). The ESP32 i2s adc input does not support stereo...

**Comment by fabioguerra-br** [comment]
Sorry, I don't ask correctly.

Using the base-adc-a2dp.ino example (https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-basic-api/base-adc-a2dp)
It read a mic connected at pin 34 (ADC) and send it to a bluetooth speaker as mono. But is possible change this to read two ADC pins (34 and 14, for example) and send to stereo bluetooth speaker?

Best regards.

**Comment by pschatzmann** [comment]
The easy answer given above is no: The ESP32 i2s adc input does not support stereo!

The more complicated answer is that it should be possible with Arduino ESP32 3.0.0-aplha 2, but you will need to [help me to have this implemented](https://github.com/pschatzmann/arduino-audio-tools/issues/1039).
